kedro-datasets/kedro_datasets/api/api_dataset.py

```diff
@@ -57,49 +56,106 @@ class APIDataSet(AbstractDataSet[None, requests.Response]):
         >>>     credentials=("username", "password")
         >>> )
         >>> data = data_set.load()
+
+    ``APIDataSet`` can also be used to save output on a remote server using
+    HTTP(S) methods.
+
+        >>> example_table = '{"col1":["val1", "val2"], "col2":["val3", "val4"]}'
+
+    Here we initialise our APIDataSet with the correct parameters to make requests
+    towards the configured remote server.
+
+        >>> data_set = APIDataSet(
+                method = "POST"
+                url = "url_of_remote_server",
+                save_args = {"chunk_size":1}
+        )
+    On initialisation, we can specify all the necessary parameters in the save args
+    dictionary. The default HTTP(S) method is POST but PUT is also supported.
+    Two important parameters to keep in mind are timeout and chunk_size. ``Timeout``
+    defines how long  our program waits for a response after a request. ``Chunk_size``, is
+    only used if the input of save method is a list. It will divide the request into
+    chunks of size ``chunk_size``. For example, here we will send two requests each
+    containing one row of our example DataFrame.
+
+        >>> data_to_save = example_table.to_dict(orient="records")
+        >>> data_set.save(data_to_save)
```

Comment:
It would be great to make sure the docstring does work when you copy and paste, I believe it's not a table anymore and `to_dict` is not needed.